docs/source/services.md

```diff
@@ -124,28 +124,162 @@ c.JupyterHub.services = [
 
 ## Writing your own services
 
-This Section Under Construction
+When writing your own services, you have a few decisions to make (in addition to what your service does!):
+
+1. Does my service need a public URL?
+2. Do I want JupyterHub to start/stop the service?
+3. Does my service need authentication?
+
+When a service is managed by JupyterHub,
+the Hub will pass the necessary information to the service via environment variables described above.
+To make your service most flexible, you can use these same environment variables,
+whether your service is managed by the Hub or not.
+
```

Comment:
When a service is managed by JupyterHub, the Hub will use the environment variables, as described above, and pass the information as needed to the service. A flexible service, whether managed by the Hub or not, can make use of these same environment variables.